src/debug/main.c

```diff
@@ -102,13 +125,20 @@ int	main(int argc, char **argv, char *env[])
 {
 	t_data	data;
 
-	signal(SIGINT, handle_sigint);
 	if (argc > 1 || argv[1])
 	{
 		ft_putstr_fd("Error: Minishell doesn't take any arguments.\n\n", 2);
 		ft_putstr_fd("Correct usage: ./minishell\n\n", 2);
 		return (0);
 	}
+	if (getenv("USER") == NULL)
+	{
+		ft_putstr_fd("Error: We truly tried but there is honestly no reliable "
+			"way of constructing the minishell prompt without the USER "
+			"environment variable set before minishell is run.\n", 2);
+		ft_putstr_fd("Please set the variable and try again.\n\n", 2);
+		return (0);
+	}
```

Comment:
There is NO reliable way to construct the minishell prompt without the USER env variable being set before minishell is launched, so if it isn't the program just won't run until it is.

The thing is that in case USER is unset during minishell running there needs to be something to fall back on to construct the user part of the prompt. The way bash does this is that it has "another copy of env somewhere in the system" that it uses in case (and actually likely always, because it has the system info always) USER is unset in the current session. It is pretty much a child (current bash session) and parent (system) process relationship where the USER gets unset only in the child, but the parent that constructs the prompt still knows the USER.
To simulate this, we need the USER to be set before launching minishell, so that we have the "parent" to fall back on.